csrc/python_frontend/fusion_record.h

```diff
@@ -379,6 +380,77 @@ struct ReshapeOpRecord : RecordFunctor {
   std::vector<int64_t> new_shape_;
 };
 
+struct PadOpRecord : RecordFunctor {
+  PadOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::vector<State>& pad_widths)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.pad",
+            RecordType::PadOp),
+        pad_widths_(std::move(pad_widths)) {}
+  virtual ~PadOpRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new PadOpRecord(*this);
+  }
+
+  virtual size_t hash() const final {
+    auto result = RecordFunctor::hash();
+    size_t widths_hash = 0;
+    // for (auto w : pad_widths_) {
+    // TODO: How to hash vector<Scalar*> pad_widths here?
+    // widths_hash ^= w.hash();
+    //}
+    return result | (widths_hash & 0xffff);
+  }
+
+  /*
+  virtual bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (auto child_ptr = dynamic_cast<const PadOpRecord*>(&other)) {
+      result = RecordFunctor::operator==(other);
+    }
```

Comment:
Wasn't quite sure how hash and equality should be defined when we have a vector of `Scalar`s as data in this record. @kevinstephano